components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -171,10 +177,36 @@ protected PostAuthnHandlerFlowStatus handlePostAuthenticationForMissingClaimsReq
                     }
                 }
 
+                List<LocalClaim> localClaims = getClaimMetadataManagementService().getLocalClaims(context.getTenantDomain());
+                Map<String, String> displayName = new HashMap<>();
+
+                for (Map.Entry<String, String> entry : missingClaimMap.entrySet()) {
+                    for (LocalClaim localClaim : localClaims){
+                        if (entry.getValue().equalsIgnoreCase(localClaim.getClaimURI())) {
+                            displayName.put(entry.getValue(), localClaim.getClaimProperties().get(DISPLAY_NAME_PROPERTY));
+                            break;
+                        }
+                    }
+                }
+
+                StringBuilder displayNameString = new StringBuilder();
+                StringBuilder displayNameValuesString = new StringBuilder();
+
+                for (Map.Entry<String, String> claimDisplay : displayName.entrySet()) {
+                    displayNameString.append(claimDisplay.getKey());
+                    displayNameValuesString.append(claimDisplay.getValue());
+                    displayNameString.append(",");
+                    displayNameValuesString.append(",");
+                }
+
+                String[] displayNames = {displayNameString.toString(), displayNameValuesString.toString()};
+
                 URIBuilder uriBuilder = new URIBuilder(ConfigurationFacade.getInstance()
                         .getAuthenticationEndpointMissingClaimsURL());
                 uriBuilder.addParameter(FrameworkConstants.MISSING_CLAIMS,
                         missingClaims[0]);
+                uriBuilder.addParameter(FrameworkConstants.DISPLAY_NAMES,
```

Comment:
Can u show an example of the URL that is built at the end.

Reply:
This is : https://accounts.asg.io/t/danushorg/authenticationendpoint/claims.do?missingClaims=http://wso2.org/claims/dob,http://wso2.org/claims/lastname,http://wso2.org/claims/givenname,&displayNames=Birth+Date,Last+Name,First+Name,&sessionDataKey=79ad3ebf-da6c-4bb3-bf4c-c439ae3077e1&sp=VinTest

![image](https://user-images.githubusercontent.com/42208518/140707789-35cc66ae-2c84-4931-8676-aeb749510384.png)

Reply:
Here there is no direct correlation between the claims and the display names. The only way is to make sure the two lists are in order. Can you check for a better solution, where the two can be properly correlated.

Reply:
According to the current implementation, it is giving display names in same order as given in missingClaims .Because it is getting display names based on mandatory missingClaims

Reply:
But ideally, the recipient of the query params should not depend on the order of two arrays to create a map of the claims and display names. This is error-prone; in case some change happens to that order.